database/sql/repositories.go

```diff
@@ -27,13 +28,12 @@ import (
 )
 
 func (s *sqlDatabase) CreateRepository(ctx context.Context, owner, name, credentialsName, webhookSecret string) (params.Repository, error) {
-	secret := []byte{}
-	var err error
-	if webhookSecret != "" {
-		secret, err = util.Aes256EncodeString(webhookSecret, s.cfg.Passphrase)
-		if err != nil {
-			return params.Repository{}, fmt.Errorf("failed to encrypt string")
-		}
+	if webhookSecret == "" {
```

Comment:
This check, and the ```webhookSecret``` checks in the other functions are good, but we should also validate that the create parameters are correct, as early as possible. A good place would be in the ```runner``` package. The ```CreateOrgParams{}```, ```CreateRepoParams{}``` and ```CreateEnterpriseParams{}```, all have a ```Validate()``` function which is called in their respective ```Create()``` functions in the ```runner``` package.

Adding a check there to see if the web hook secret is set, would be nice.

Reply:
I added the checks in the validation funcs. Don't you think that we can remove all other checks later on?

Reply:
I usually don't do much validation in the store layer on things that should be validated elsewhere. Unless there is an error when trying to encode an empty string, we can remove this check. We need to trust that the data the user wants to save is correct. The store's job is to save and retrieve the data we want. It has no knowledge of the correctness of that data.

So you're right, this can be removed.